File: week_2/task_b.cc
#include <iostream>
#include <string>

int main() {
  std::string river;
  std::cin >> river;

  int l_min = 0, r_min = 1;
  for (char c : river) {
    if (c == 'L') {
      r_min = std::min(r_min, l_min + 1);
      l_min = std::min(l_min + 1, r_min + 1);
    } else if (c == 'R') {
      l_min = std::min(l_min, r_min + 1);
      r_min = std::min(r_min + 1, l_min + 1);
    } else {
      ++l_min;
      ++r_min;
    }
  }

  std::cout << std::min(l_min + 1, r_min);

  return 0;
}

File: week_2/go/task_b/task_b.go
package main

import "fmt"

func main() {
	var river string

	fmt.Scan(&river)

	var l_min, r_min int

	r_min = 1

	for _, c := range river {
		switch c {
		case 'L':
			r_min = min(r_min, l_min+1)
			l_min = min(l_min+1, r_min+1)
		case 'R':
			l_min = min(l_min, r_min+1)
			r_min = min(r_min+1, l_min+1)
		default:
			l_min++
			r_min++
		}
	}

	fmt.Println(min(l_min+1, r_min))
}
